src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -12,8 +13,17 @@ namespace System.Net.NetworkInformation
     // Linux implementation of NetworkChange
     public class NetworkChange
     {
-        private static NetworkAddressChangedEventHandler s_addressChangedSubscribers;
-        private static NetworkAvailabilityChangedEventHandler s_availabilityChangedSubscribers;
+        private static readonly Dictionary<NetworkAddressChangedEventHandler, ExecutionContext> s_addressChangedSubscribers =
+            new Dictionary<NetworkAddressChangedEventHandler, ExecutionContext>();
+        private static readonly Dictionary<NetworkAvailabilityChangedEventHandler, ExecutionContext> s_availabilityChangedSubscribers =
+            new Dictionary<NetworkAvailabilityChangedEventHandler, ExecutionContext>();
+
+        private static readonly NetworkAvailabilityEventArgs s_availableEventArgs = new NetworkAvailabilityEventArgs(isAvailable: true);
+        private static readonly NetworkAvailabilityEventArgs s_notAvailableEventArgs = new NetworkAvailabilityEventArgs(isAvailable: false);
+        private static readonly ContextCallback s_runHandlerAvailable = new ContextCallback(RunAvailabilityHandlerAvailable);
+        private static readonly ContextCallback s_runHandlerNotAvailable = new ContextCallback(RunAvailabilityHandlerNotAvailable);
+        private static readonly ContextCallback s_runAddressChangedHandler = new ContextCallback(RunAddressChangedHandler);
```

Comment:
This is adding a fair amount of duplication between the Linux and OSX implementations, and it's also missing this optimization for the Windows implementation.  Can you move this shared code into a shared NetworkAddressChange.cs partial file, and then also use these s_{not}AvailableEventArgs from the Windows implementation?  More can be deduped later, but we should at least avoid adding more duplication.

Reply:
Deduped; also copied dictionaries under lock for Linux/OSX and skipped trying to add null listeners as per Windows